integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -246,6 +247,126 @@ object BenchUtils {
     }
   }
 
+  /**
+   * Perform a diff of the results collected from two DataFrames, allowing for differences in
+   * precision.
+   *
+   * This is only suitable for data sets that can fit in the driver's memory.
+
+   * @param df1 DataFrame to compare.
+   * @param df2 DataFrame to compare.
+   * @param ignoreOrdering Sort the data collected from the DataFrames before comparing them.
+   * @param epsilon Allow for differences in precision when comparing floating point values.
+   */
+  def compareResults(
+      df1: DataFrame,
+      df2: DataFrame,
+      ignoreOrdering: Boolean,
+      maxErrors: Int = 10,
+      epsilon: Double = 0.00001): Unit = {
+
+    val result1: Seq[Seq[Any]] = collectResults(df1, ignoreOrdering)
+    val result2: Seq[Seq[Any]] = collectResults(df2, ignoreOrdering)
+
+    if (result1.length == result2.length) {
+      var errors = 0
+      var i = 0;
+      while (i < result1.length && errors < maxErrors) {
+        val l = result1(i)
+        val r = result2(i)
+        if (!rowEqual(l, r, epsilon)) {
+          println(s"Row $i:\n${l.mkString(",")}\n${r.mkString(",")}\n")
+          errors += 1
+          if (errors == maxErrors) {
+            println(s"Aborting comparison after reaching maximum of $maxErrors errors")
+          }
+        }
+        i += 1
+      }
+
+      if (errors==0) {
+        println(s"Results match")
+      }
+
+    } else {
+      println(s"Row counts do not match: ${result1.length} != ${result2.length}")
+    }
+  }
+
+  private def collectResults(df: DataFrame, ignoreOrdering: Boolean): Seq[Seq[Any]] = {
+    println("Collecting rows from DataFrame")
+    val t1 = System.currentTimeMillis()
+    val rows = if (ignoreOrdering) {
+      // let Spark do the sorting
+      df.sort(df.columns.map(col): _*).collect()
```

Comment:
the downside to this is if sort is broken, is this forcing cpu sort for instance?

Reply:
The only known issue with GPU sort vs CPU sort is #84 Which I don't think we will ever run into outside of artificially generated data sets, which is why we have not pushed to fix it.

Reply:
> the downside to this is if sort is broken, is this forcing cpu sort for instance?

The job config is outside of the tool. I think to verify the results we should let standard cpu spark do it, nothing stops you from running this with the plugin, so it's up to the user at this point.

Reply:
I guess we have other tests for correctness so ignore my comment, and hopefully the GPU sort should be faster.